### PR TITLE
Multiply method

### PR DESCRIPTION
**Pull request by MarcAntoineSchmidtQC**
This PR adds a `multiply` method ~~and the `__mul__` and `__rmul__` special methods~~ to the matrices. This is needed for [this glum PR](https://github.com/Quantco/glum/pull/526).

Checklist
* [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

File: src/tabmat/categorical_matrix.py [hunk 1]
```diff
@@ -597,38 +597,46 @@ def _cross_sparse(
...
 
         res = term_1.T.dot(_row_col_indexing(other, rows, R_cols)).A
         return res
 
-    def multiply(self, other) -> sps.csr_matrix:
-        """Element-wise multiplication of each column with other."""
+    def multiply(self, other) -> SparseMatrix:
+        """Element-wise multiplication.
+
+        This assumes that ``other`` is a vector of size self.shape[1].
```
Comment (KrzysztofChomiakQC) [inline src/tabmat/categorical_matrix.py:610]: This line of docstring seems inconsistent with line 612 below, where the check is performed with ``self.shape[0]`` not on ``self.shape[1]``

Comment (MarcAntoineSchmidtQC) [reply src/tabmat/categorical_matrix.py:610]: Thanks! 🤦

**Comment by DavidMeesterQC** [comment]
Thanks! Looks good to me 🚀

File: src/tabmat/categorical_matrix.py [hunk 2]
```diff
@@ -597,38 +597,46 @@ def _cross_sparse(
...
 
         res = term_1.T.dot(_row_col_indexing(other, rows, R_cols)).A
         return res
 
-    def multiply(self, other) -> sps.csr_matrix:
-        """Element-wise multiplication of each column with other."""
+    def multiply(self, other) -> SparseMatrix:
+        """Element-wise multiplication.
+
+        This assumes that ``other`` is a vector of size self.shape[0].
```
Comment (lbittarello) [inline src/tabmat/categorical_matrix.py:610]:
```suggestion
        This assumes that ``other`` is a vector of size ``self.shape[0]``.
```
?

**Review by jtilly — APPROVED** [review]

**Pull request merged**